lib/active_model/serializer/builder.rb

```diff
@@ -0,0 +1,71 @@
+require "set"
+module ActiveModel
+  class Serializer
+    class Builder
```

Comment:
So, I think this PR is a great step in organizing things. The current serializer is a huge mess.

With that said, I just want to leave registered in the interwebs a comment here about nomenclatures that we think we need to improve, if not in this PR, in the future. I think it's a bad idea to objects/classes with anything ending in `*er` because you'll end up with things like `Builder#build`. `SerializedHash#build` expresses much better its intent. I've written about it and more in http://alexsquest.com/texts/complexity-in-software-4-the-art-of-naming. There's also http://www.carlopescio.com/2011/04/your-coding-conventions-are-hurting-you.html.

Not a blocker, just something to improve, perhaps in another PR.